src/test/java/it/eng/idsa/dataapp/util/MessageUtilTest.java

```diff
@@ -0,0 +1,150 @@
+package it.eng.idsa.dataapp.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.HttpHeaders;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import de.fraunhofer.iais.eis.Connector;
+import de.fraunhofer.iais.eis.DescriptionRequestMessage;
+import de.fraunhofer.iais.eis.RejectionMessage;
+import de.fraunhofer.iais.eis.Resource;
+import de.fraunhofer.iais.eis.ids.jsonld.Serializer;
+import it.eng.idsa.dataapp.configuration.ECCProperties;
+import it.eng.idsa.dataapp.service.MultiPartMessageService;
+import it.eng.idsa.multipart.processor.util.SelfDescriptionUtil;
+import it.eng.idsa.multipart.processor.util.TestUtilMessageService;
+
+public class MessageUtilTest {
+	
+	private MessageUtil messageUtil;
+	
+	@Mock
+	private Path dataLakeDirectory;
+	
+	@Mock
+	private RestTemplate restTemplate;
+	
+	@Mock
+	private ECCProperties eccProperties;
+	
+	@Mock
+	private MultiPartMessageService multiPartMessageService;
+	
+	private HttpHeaders headers;
+	
+	private Serializer serializer = new Serializer();
+	
+	private static final String IDS_MESSAGE_TYPE = "IDS-MessageType";
+	private static final String IDS_REQUESTED_ELEMENT = "IDS-RequestedElement";
+	private static final URI EXISTING_REQUESTED_ELEMENT_ID = SelfDescriptionUtil.getBaseConnector().getResourceCatalog().get(0).getOfferedResource().get(0).getId();
+	private static final URI NON_EXISTING_REQUESTED_ELEMENT_ID = URI.create(EXISTING_REQUESTED_ELEMENT_ID + "NonExistingElement");
+	
+	@BeforeEach
+	public void init() throws RestClientException, IOException {
+		MockitoAnnotations.initMocks(this);
+		when(eccProperties.getHost()).thenReturn("localhost");
+		when(restTemplate.getForObject(any(), any())).thenReturn(serializer.serialize(SelfDescriptionUtil.getBaseConnector()));
+		when(multiPartMessageService.createRejectionCommunicationLocalIssues(any())).thenReturn(TestUtilMessageService.getRejectionMessage());
+		dataLakeDirectory = Path.of("/dataLakeDirectory");
+		messageUtil = new MessageUtil(dataLakeDirectory, restTemplate, eccProperties, multiPartMessageService);
+		headers = new HttpHeaders();
+	}
+	
+	@Test
+	public void createResponsePayloadFromDescriptionRequestMessageWithoutRequestedElementMethodParameterMessageSuccessfull() throws IOException {
```

Comment:
Can we make those method names more readable?
6 tests have createResponsePayloadFromDescriptionRequestMessageWithoutRequestedElementMethodParameter + something else

Next are createResponsePayloadFromDescriptionRequestMessageWithRequestedElementMethodParameter + something.

testResponsePayloadWithoutRequestedElement + something for example, or even more simple. Feel free to use comments to separate some use cases.

Reply:
Sure, I'll figure something out.